### DISCUSSION TITLE
SEB splash screen issues

### DISCUSSION THREAD

**Discussion by VirtualRusty** [SafeExamBrowser/seb-mac#319] [General]
I work at a university and Inspera recently upgraded to allow version 3 of Mac SEB, so we needed to check the version of SEB of Macs before students enter the exam room. We can easily have about 300 mac users entering the exam rooms from the foyer each session. As part of entry they have to show that they've logged in to SSO > Inspera and be at the exam start password prompt. This cuts down on errors we have to deal with inside the room by flagging issues in the foyer and sending them to the IT desk there. However, there are two issues with macs that make this difficult. One is that for a percentage of mac users the screen is completely black when the password popup appears, so we can't see the version number or anything else, and for new staff it looks like a major fault. The second is the password popup obscures the version number on the Mac (but not Win SEB). I thought I had screen captured this error but can't find the photo, but would it be possible in a future version to have the exam password popup a bit lower on the screen so we can read the version number? At the moment we have to ask the student to move the popup which is tedious. I've included a shot of the black screen.

![IMG_050473](https://github.com/SafeExamBrowser/seb-mac/assets/120538228/b523af0e-ae8a-4b7f-9e4c-1482647ba303)

This is the splash screen we need to read but its obscured when the password popup is launched

![IMG_050479 copy](https://github.com/SafeExamBrowser/seb-mac/assets/120538228/1c6b7ab8-b912-47d8-b061-6eba4e5a58de)

**Comment by danschlet** [comment]
Your use case is not supported by SEB. You will have to change it or wait approximately until next year when we have time to deal with such minor issues (we have a lot of major tasks and features to implement with high priority).

When starting an exam, the assessment system can check which SEB version is used and show a warning or dialog to the user to ask them to update to a newer version of SEB. You could create a simple web app or exam portal page which checks the user agent string and informs students accordingly. Then you could direct them to use the URL of that exam portal to start their exams (you would redirect to the actual assessment app like Inspera from there).

Also we specified new settings which will allow to preset with which SEB version(s) an exam configuration will work. When a non-compliant SEB version opens such settings (but after decrypting it, so only after the exam password prompt you mentioned), a dialog will be displayed to update to a newer SEB version and then restart the exam. If I'm not wrong this setting will be available from SEB 3.6 Windows and 3.4 Mac/iOS.

That a black screen is displayed on some Macs in the moment when the exam password prompt is displayed while others already display a browser window and the splash screen is a macOS concurrency timing issue and most likely due to different timing of the activation of the SEB lockdown (kiosk) mode on different macOS versions and Mac models. As long as starting the exam works, this is nothing we can spend time on improving. Current versions of SEB might anyways behave differently, as a reminder we don't support SEB 2.3.2 anymore, you should try such issues with the latest SEB version before reporting it.

**Reply by VirtualRusty** [reply]
Thanks @danschlet. Unfortunately 20%+ of real world students don't follow any instructions so that's why we implemented the "no entry unless displaying password prompt and version number" on an open laptop. This rule reduced in-the-room start of exam chaos by 80%. Version 2.x.x showed the version number when the password prompt displayed, unfortunately the obscuration issue appeared in version 3. I included the screenshot of 2.3.2 as an example of what was obscured that we were trying to identify. If we see that we send them to the helpdesk.

Many students are not technically literate and they will click the download button below "your version is out of date" for the latest SEB and believe that it will install automatically. Its not uncommon for me to see 4-6 duplicate files of the current version in the downloads folder as they keep clicking the download link.

I understand this issue can't be fixed in an instant but it seems to be a simple positioning issue, and I hope at some time it can be fixed.

**Reply by danschlet** [reply]
I added the SEB version to the Starting Exam prompt window title, maybe that will help resolving your issue, it is available in [Release Candidate 2 of SEB 3.3](https://github.com/SafeExamBrowser/seb-mac/releases/tag/3.3rc2).

I still would feel very uneasy having to walk into an exam hall with my laptop's lid open. By using such an exam portal (which URL would need to be the sole exam entry point communicated to students) you could prevent UX issues like with the download button below "your version is out of date" in Inspera Assessment by providing your own customized instructions if students attempt to open that portal page with an outdated SEB version. You would need to use separate SEB settings (with the allow reconfiguring option set to be able to open the exam system from there, also probably disable the clear session cookies on session end setting) to open the portal page in SEB, those wouldn't have to set a quit password so students couldn't get locked into SEB if they need to restart their computer before having started the exam. Another option which works for us is to require new students to open **and submit** a mock exam in SEB before even receiving the confirmed date/place for their real exam.

**Reply by danschlet** [reply]
<img width="461" alt="Screenshot 2023-08-26 at 22 59 34" src="https://github.com/SafeExamBrowser/seb-mac/assets/2464749/c5362ef1-8173-4c8d-952a-1fe01b6f482d">

**Reply by VirtualRusty** [reply]
thanks @danschlet that's a really good all-in-one solution and we can see immediately what version it is. Some students voluntarily do the practice exam before entry then wonder why the password doesn't work, not realising they are still in the test exam, not the exam they are sitting. Your suggestions re downloading the latest versions are solid and we do run those server-side version checks and prompts, but so many students miss or don't understand these prompts. On the plus side, students doing their third and forth exam are learning what to do in advance and teaching the friends so its reducing our workload to a degree. But again, appreciate this improvement.

**Reply by danschlet** [reply]
Also thank you for your input. It's good for us to learn about specific requirements which we didn't think of, because our own SEB uses cases are different. Our first reaction might be to find a workaround (as we have a large backlog of feature requests), but I'm glad if we can find a quick working solution like in this case.

I hope at some point we'll have capacity to look into a SEB version check with direct or indirect update option. It would still cause delays if an update is prompted at the beginning of an exam, but if for example SEB then would directly download the new version and start the updater, that would probably make it easier to manage a (optionally required) update for students. We just would try to avoid having to run an "update server" but would just use GitHub Releases and not perform an automatic version check by default, but only if explicitely requested in the used SEB configuration.

**Reply by VirtualRusty** [reply]
Direct update would work for us as our demand that students log in to the password prompt before entering the exam room means that all downloads/update installs would be done in the foyer, not in the exam room, so they can start the exam without interruption. Would make our job a lot easier not having the endlessly walk the foyer pointing out where the downloaded update is and how to install it :-)
